rust/transaction_parsing/src/check_signature.rs

```diff
@@ -26,9 +26,10 @@ pub fn pass_crypto(
             // Ed25519 crypto was used by the verifier
             let (pubkey, data) = match data.get(3..35) {
                 Some(a) => {
-                    let into_pubkey: [u8; 32] =
-                        a.try_into().expect("fixed size should fit in array");
-                    (ed25519::Public::from_raw(into_pubkey), &data[35..])
+                    let into_pubkey = H256::from(
```

Comment:
These are public keys, H256 should not be here, as there is no hashing. Just below is `ecdsa` with public key length 33.